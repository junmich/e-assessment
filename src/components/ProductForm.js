import React, { useState, useEffect } from 'react';

const ProductForm = ({ handleCreateUpdateProduct, product, backButton }) => {
    const [productName, setProductName] = useState('');
    const [description, setDescription] = useState('');
    const [error, setError] = useState(undefined);

    useEffect(() => {
        if (product) {
            setProductName(product.name);
            setDescription(product.description);

        }
    }, [product]);
    const clearForm = () => {
        setProductName('');
        setDescription('');
        setError('');
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        const type = product ? 'edit' : 'add';
        const errorName = handleCreateUpdateProduct({ name: productName, description }, type);
        if (!errorName) {
            clearForm();
        } else {
            setError(errorName);
        }            
    };

    return (
        <div className="product-form">
            {error && <div className="error-message">{error}</div>}
            <form onSubmit={handleSubmit}>
                <div>
                    <input 
                        className="product-form__input"
                        readOnly={product} value={productName} 
                        onChange={e => setProductName(e.target.value)} 
                        type="text"
                    />
                </div>
                <div>
                    <textarea 
                        className="product-form__text-area" 
                        onChange={e => setDescription(e.target.value)} 
                        value={description} 
                    />
                </div>
                <button 
                    className="default-button" 
                    type="submit">
                        {product? 'Update' : 'Add'}
                </button>
                <button
                    onClick={backButton}
                    className="delete-button" 
                    type="button">
                        Back
                </button>
            </form>
        </div>
    );
}


export default ProductForm;