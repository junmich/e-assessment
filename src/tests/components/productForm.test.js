import React from 'react';
import { render } from 'enzyme';
import ProductForm from '../../components/ProductForm';

describe('Product Form Test', () => {
    const handleCreateUpdateProduct = jest.fn();
    const backButton = jest.fn();
    test('should render product form correctly', () => {
        const wrapper = render(<ProductForm handleCreateUpdateProduct={handleCreateUpdateProduct} backButton={backButton} />);
        expect(wrapper.find('.product-form__input').length).toBe(1);
        expect(wrapper.find('.default-button').length).toBe(1);
        expect(wrapper.find('.delete-button').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });
});

