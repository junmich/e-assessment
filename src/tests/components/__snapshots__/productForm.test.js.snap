// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Form Test should render product form correctly 1`] = `
<div
  class="product-form"
>
  <form>
    <div>
      <input
        class="product-form__input"
        type="text"
        value=""
      >
        
      </input>
    </div>
    <div>
      <textarea
        class="product-form__text-area"
      >
        
      </textarea>
    </div>
    <button
      class="default-button"
      type="submit"
    >
      Add
    </button>
    <button
      class="delete-button"
      type="button"
    >
      Back
    </button>
  </form>
</div>
`;
